### PR TITLE
UCS/CONFIG: fixed crash on incorrect value set

### PR DESCRIPTION
**Pull request by hoopoepg**
- there was possible double free of string when
  user tried to set incorrect value using ucs_config_parser_set_value
  function

fixes https://github.com/openucx/ucx/issues/5980

### CODE REVIEW

**Review by yosefe — COMMENTED** [review]
pls add gtest

**Comment by hoopoepg** [comment]
added test

**Comment by yosefe** [comment]
bot:pipe:retest

File: test/gtest/ucs/test_config.cc [hunk 1]
```diff
@@ -457,6 +481,14 @@ UCS_TEST_F(test_config, set_get) {
 
     opts.set("VIN", "123456");
     EXPECT_EQ(123456UL, opts->vin);
+
+    /* try to set incorrect value - color should not be updated */
+    scoped_log_handler log_handler_vars(config_error_suppress);
+    opts.set("COLOR", "magenta");
```
Comment (yosefe) [inline test/gtest/ucs/test_config.cc:487]: pls wrap these in scope { }

File: test/gtest/ucs/test_config.cc [hunk 2]
```diff
@@ -246,6 +253,21 @@ class test_config : public ucs::test {
         return UCS_LOG_FUNC_RC_CONTINUE;
     }
 
+    static ucs_log_func_rc_t
+    config_error_suppress(const char *file, unsigned line, const char *function,
```
Comment (yosefe) [inline test/gtest/ucs/test_config.cc:257]: can reuse wrap_errors_logger ?

Comment (hoopoepg) [reply test/gtest/ucs/test_config.cc:257]: technically yes, but in this case we can't check if error is handled - there is no errors counter in ```wrap_errors_logger```

File: test/gtest/ucs/test_config.cc (hunk 1 above)
Comment (hoopoepg) [reply test/gtest/ucs/test_config.cc]: wrapped by {}

File: test/gtest/ucs/test_config.cc (hunk 2 above)
Comment (yosefe) [reply test/gtest/ucs/test_config.cc:257]: can we call wrap_errors_logger() after counting, to print err message?

File: test/gtest/ucs/test_config.cc [hunk 3]
```diff
@@ -483,12 +483,14 @@ UCS_TEST_F(test_config, set_get) {
...
-    EXPECT_EQ(std::string(color_names[COLOR_WHITE]),
-              std::string(opts.get("COLOR")));
-    EXPECT_EQ(1, m_errors);
+    {
+        scoped_log_handler log_handler_vars(config_error_suppress);
+        opts.set("COLOR", "magenta");
+        EXPECT_EQ(COLOR_WHITE, opts->color);
+        EXPECT_EQ(std::string(color_names[COLOR_WHITE]),
+                std::string(opts.get("COLOR")));
+        EXPECT_EQ(1, m_errors);
```
Comment (yosefe) [inline test/gtest/ucs/test_config.cc:492]: these can be outside the scope

File: test/gtest/ucs/test_config.cc (hunk 2 above)
Comment (hoopoepg) [reply test/gtest/ucs/test_config.cc:257]: ok, wrapped

File: test/gtest/ucs/test_config.cc (hunk 3 above)
Comment (hoopoepg) [reply test/gtest/ucs/test_config.cc]: moved outside of scope

File: test/gtest/ucs/test_config.cc [hunk 4]
```diff
@@ -246,6 +253,23 @@ class test_config : public ucs::test {
...
+                          ucs_log_level_t level,
+                          const ucs_log_component_config_t *comp_conf,
+                          const char *message, va_list ap)
+    {
+        // Ignore errors that invalid input parameters as it is expected
+        if (level == UCS_LOG_LEVEL_ERROR) {
+            m_errors++;
+            wrap_errors_logger(file, line, function, level, comp_conf,
+                               message, ap);
+            return UCS_LOG_FUNC_RC_STOP;
```
Comment (yosefe) [inline test/gtest/ucs/test_config.cc:267]: minor: return wrap_errors_logger(..)

File: test/gtest/ucs/test_config.cc [hunk 5]
```diff
@@ -224,7 +224,14 @@ ucs_config_field_t car_opts_table[] = {
 static std::vector<std::string> config_err_exp_str;
 
 class test_config : public ucs::test {
+public:
+    test_config() {
+        m_errors = 0;
```
Comment (yosefe) [inline test/gtest/ucs/test_config.cc:229]: minor: m_num_errors

Comment (hoopoepg) [reply test/gtest/ucs/test_config.cc]: renamed to m_num_errors

File: test/gtest/ucs/test_config.cc (hunk 4 above)
Comment (hoopoepg) [reply test/gtest/ucs/test_config.cc]: updated

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@hoopoepg pls squash

**Comment by hoopoepg** [comment]
squashed

**Pull request merged**